xarray/core/variable.py

```diff
@@ -739,7 +739,7 @@ def _getitem_with_mask(self, key, fill_value=dtypes.NA):
 
             data = as_indexable(self._data)[actual_indexer]
             mask = indexing.create_mask(indexer, self.shape, data)
-            data = duck_array_ops.where(mask, fill_value, data)
+            data = duck_array_ops.where(~mask, data, fill_value)
```

Comment:
While trying to understand this change, the comments on [this StackOverflow post](https://stackoverflow.com/questions/8305199/the-tilde-operator-in-python) made me wonder if using the tilde operator here could produce different results when `mask` is a bool vs. when it is a np.ndarray of dtype bool (or other return types of [create_mask](https://github.com/pydata/xarray/blob/f468a0676b2411a53e4bb94d67918092d06b50b8/xarray/core/indexing.py#L1129-L1147)).

Reply:
yes, that's true. In #3706 we agreed on using `numpy.logical_not` instead.

This is code from a long time ago, I really need to update / rebase this PR.